test/framework/src/main/java/org/elasticsearch/test/rest/RestTestLegacyFeatures.java

```diff
@@ -23,29 +24,61 @@
  */
 public class RestTestLegacyFeatures implements FeatureSpecification {
     public static final NodeFeature ML_STATE_RESET_FALLBACK_ON_DISABLED = new NodeFeature("ml.state_reset_fallback_on_disabled");
+    @UpdateForV9
     public static final NodeFeature FEATURE_STATE_RESET_SUPPORTED = new NodeFeature("system_indices.feature_state_reset_supported");
     public static final NodeFeature SYSTEM_INDICES_REST_ACCESS_ENFORCED = new NodeFeature("system_indices.rest_access_enforced");
+    @UpdateForV9
+    public static final NodeFeature SYSTEM_INDICES_REST_ACCESS_DEPRECATED = new NodeFeature("system_indices.rest_access_deprecated");
```

Comment:
Is there an implied ordering in these NodeFeatures? If not, should we consider ordering them by historical version number?

Just a thought; no need to address it for this PR in particular.

Reply:
It's a very good point; so far they are (roughly) grouped by usage: core test framework, rolling-upgrade, YAML. But it's a loose order; once I have done (or I am at a reasonable point), I might reorder them or even split them across multiple FeatureSpecifications.